### PR TITLE
download timeouts

### PR DESCRIPTION
**Pull request by MartinNowak**
- 8s by default
- only for curl atm.

### CODE REVIEW

File: test/timeout.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Server: dummy\r
+Content-Type: application/json\r
+Content-Length: 2\r
+\r
+{}')
+    for i in $(seq 0 $((${#res} - 1))); do
+        echo -n "${res:$i:1}"
+        sleep 1
+    done
+} | nc --listen $PORT >/dev/null &
```
Comment (MartinNowak) [inline test/timeout.sh]: terminal

File: test/timeout.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+. $(dirname "${BASH_SOURCE[0]}")/common.sh
+
+PORT=$(($$ + 1024)) # PID + 1024
```
Comment (wilzbach) [inline test/timeout.sh]: Good, old std.socket can get you a free port too:

```
rdmd --eval='import std.socket; auto s = new Socket(AddressFamily.INET, SocketType.STREAM); scope(exit) s.close; s.bind(new InternetAddress(0));s.localAddress.toPortString.writeln;'
```

**Comment by MartinNowak** [comment]
> I like the tests!

Would be nicer if it were unittests in the ms range, but that was too much work.

File: test/timeout.sh (hunk 2 above)
Comment (MartinNowak) [reply test/timeout.sh]: It's fine there won't be colliding PIDs.

**Comment by MartinNowak** [comment]
Merging this as it's crucial for the mirror mechanism to actually work. We had to enjoy the recent code.dlang.org downtime party to notice ;). It'll be out with 2.078.1 soon, so we win some time to address stability.

**Pull request merged**